### ISSUE TITLE
Avoid CloudFront invalidation when there are no changes

### ISSUE THREAD

**Issue by jponge** [laurilehmijoki/s3_website#149]
It seems that `s3_website` performs CloudFront invalidation even when there is nothing to push:

```console
$ s3_website push --dry-run
[info] Simulating the deployment of /<SNIP>/_site/* to <SNIP>
[succ] Would have invalidated 1 item on CloudFront
[info] Summary: Would have pushed nothing
$
```

This should not be the case IMHO.

**Comment by laurilehmijoki** [comment]
I can confirm that this bug exists. Let's fix it.

**Comment by laurilehmijoki** [comment]
I just released the version **2.8.3** where this problem should be fixed. 

Can you confirm that the fix works for you?

**Comment by jponge** [comment]
All good :+1:
